From 43d153dd7ed2d03990ae1849a9adf215d7231e1c Mon Sep 17 00:00:00 2001
From: LinsaFTW <25271111+linsaftw@users.noreply.github.com>
Date: Tue, 15 Mar 2022 11:29:43 -0300
Subject: [PATCH] Adaptative chunk GC


diff --git a/src/main/java/org/bukkit/craftbukkit/CraftWorld.java b/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
index 584070c0..1df845f2 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
@@ -1393,7 +1393,14 @@ public class CraftWorld implements World {
     public void processChunkGC() {
         chunkGCTickCount++;
 
-        if (chunkLoadCount >= server.chunkGCLoadThresh && server.chunkGCLoadThresh > 0) {
+        // FlamePaper start - Adaptative chunk GC
+        int playerCount = getPlayers().size();
+        int viewDistance = getHandle().getServer().getViewDistance();
+        int viewChunks = ((viewDistance * 2) + 1);
+        int chunkGCLoadThreshold = org.github.paperspigot.PaperSpigotConfig.adaptativeChunkGC ? (world.keepSpawnInMemory ? 256 : 0) + playerCount * (viewChunks * viewChunks) : server.chunkGCLoadThresh;
+
+        if (chunkLoadCount > chunkGCLoadThreshold && chunkGCLoadThreshold > 0) { 
+        // FlamePaper end - Adaptative chunk GC
             chunkLoadCount = 0;
         } else if (chunkGCTickCount >= server.chunkGCPeriod && server.chunkGCPeriod > 0) {
             chunkGCTickCount = 0;
diff --git a/src/main/java/org/github/paperspigot/PaperSpigotConfig.java b/src/main/java/org/github/paperspigot/PaperSpigotConfig.java
index 2153682f..0f5bb5f2 100644
--- a/src/main/java/org/github/paperspigot/PaperSpigotConfig.java
+++ b/src/main/java/org/github/paperspigot/PaperSpigotConfig.java
@@ -186,6 +186,12 @@ public class PaperSpigotConfig
         bookMaxPages = getInt("book.max_pages", 5);
     }
     
+    public static boolean adaptativeChunkGC;
+    private static void adaptativeChunkGC()
+    {
+        adaptativeChunkGC = getBoolean("adaptative-chunk-gc", true);
+    }
+
     public static boolean allowMapDecorations;
     private static void allowMapDecorations()
     {
-- 
2.35.3.windows.1

