From 3808f45c3dbfbf1a7ca12e9bc99b23ffefc74411 Mon Sep 17 00:00:00 2001
From: LinsaFTW <25271111+linsaftw@users.noreply.github.com>
Date: Sun, 13 Mar 2022 22:56:52 -0300
Subject: [PATCH] Disable Unloaded Chunk Movement


diff --git a/src/main/java/net/minecraft/server/Entity.java b/src/main/java/net/minecraft/server/Entity.java
index c4bfd5fe..79c31a65 100644
--- a/src/main/java/net/minecraft/server/Entity.java
+++ b/src/main/java/net/minecraft/server/Entity.java
@@ -432,6 +432,12 @@ public abstract class Entity implements ICommandListener {
 
     public void move(double d0, double d1, double d2) {
         if (this.loadChunks) loadChunks(); // PaperSpigot - Load chunks
+        // FlamePaper start - Disable Unloaded Chunk Movement
+        if (!((ChunkProviderServer) world.chunkProvider).isChunkLoaded((int) locX >> 4, (int) locZ >> 4)) {
+            this.a(this.getBoundingBox().c(d0, d1, d2));
+            this.recalcPosition();   
+        } else
+        // FlamePaper end - Disable Unloaded Chunk Movement
         if (this.noclip) {
             this.a(this.getBoundingBox().c(d0, d1, d2));
             this.recalcPosition();
-- 
2.35.3.windows.1

