From 22c8fa8cfae1183e687f733dad9e6b9a040087e3 Mon Sep 17 00:00:00 2001
From: LinsaFTW <25271111+linsaftw@users.noreply.github.com>
Date: Mon, 14 Mar 2022 15:34:33 -0300
Subject: [PATCH] Disable Nether Roof Interaction


diff --git a/src/main/java/net/minecraft/server/ItemStack.java b/src/main/java/net/minecraft/server/ItemStack.java
index 015ce283..93c453d2 100644
--- a/src/main/java/net/minecraft/server/ItemStack.java
+++ b/src/main/java/net/minecraft/server/ItemStack.java
@@ -154,6 +154,25 @@ public final class ItemStack {
             org.bukkit.event.block.BlockPlaceEvent placeEvent = null;
             List<BlockState> blocks = (List<BlockState>) world.capturedBlockStates.clone();
             world.capturedBlockStates.clear();
+
+            // FlamePaper start - Disable Nether Roof Interaction
+            if (world.getWorld().getEnvironment() == org.bukkit.World.Environment.NETHER) {
+                for (BlockState block : blocks) {
+                    if (block.getY() >= 127) {
+                        entityhuman.getBukkitEntity().sendMessage(org.bukkit.ChatColor.translateAlternateColorCodes('&', "&cCan't build on nether roof"));
+                        flag = false; // cancel placement
+                        // revert back all captured blocks
+                        for (BlockState blockstate : blocks) {
+                            blockstate.update(true, false);
+                        }
+                        world.capturedTileEntities.clear();
+                        world.capturedBlockStates.clear();
+                        return flag;
+                    }
+                }
+            }
+            // FlamePaper end - Disable Nether Roof Interaction
+            
             if (blocks.size() > 1) {
                 placeEvent = org.bukkit.craftbukkit.event.CraftEventFactory.callBlockMultiPlaceEvent(world, entityhuman, blocks, blockposition.getX(), blockposition.getY(), blockposition.getZ());
             } else if (blocks.size() == 1) {
diff --git a/src/main/java/net/minecraft/server/PlayerInteractManager.java b/src/main/java/net/minecraft/server/PlayerInteractManager.java
index 33a0a095..3bb579e7 100644
--- a/src/main/java/net/minecraft/server/PlayerInteractManager.java
+++ b/src/main/java/net/minecraft/server/PlayerInteractManager.java
@@ -251,6 +251,15 @@ public class PlayerInteractManager {
         BlockBreakEvent event = null;
 
         if (this.player instanceof EntityPlayer) {
+            // FlamePaper start - Disable Nether Roof Interaction
+            if (world.getWorld().getEnvironment() == org.bukkit.World.Environment.NETHER &&
+                blockposition.getY() >= 127) {
+                this.player.getBukkitEntity().sendMessage(org.bukkit.ChatColor.translateAlternateColorCodes('&', "&cCan't build on nether roof"));
+                ((EntityPlayer) this.player).playerConnection.sendPacket(new PacketPlayOutBlockChange(this.world, blockposition));
+                return false;
+            }
+            // FlamePaper end - Disable Nether Roof Interaction
+
             org.bukkit.block.Block block = this.world.getWorld().getBlockAt(blockposition.getX(), blockposition.getY(), blockposition.getZ());
 
             // Sword + Creative mode pre-cancel
-- 
2.35.3.windows.1

