From c895b9e6989b41428f1e42b0bf4ad56178e7193b Mon Sep 17 00:00:00 2001
From: LinsaFTW <25271111+linsaftw@users.noreply.github.com>
Date: Wed, 25 Jan 2023 13:54:24 -0300
Subject: [PATCH] Supress Invalid Book Exceptions


diff --git a/src/main/java/net/minecraft/server/PlayerConnection.java b/src/main/java/net/minecraft/server/PlayerConnection.java
index 0c862db8..cd81a051 100644
--- a/src/main/java/net/minecraft/server/PlayerConnection.java
+++ b/src/main/java/net/minecraft/server/PlayerConnection.java
@@ -2041,7 +2041,7 @@ public class PlayerConnection implements PacketListenerPlayIn, IUpdatePlayerList
                     return;
                 }
             } catch (Exception exception1) {
-                PlayerConnection.c.error("Couldn\'t sign book", exception1);
+                // FlamePaper - Supress Invalid Book Exceptions
                 this.disconnect("Invalid book data!"); // CraftBukkit
                 return;
             } finally {
-- 
2.35.3.windows.1

