From 05000976ac46a3a89fe5ccdb6f6b89ebec68a57e Mon Sep 17 00:00:00 2001
From: LinsaFTW <25271111+linsaftw@users.noreply.github.com>
Date: Thu, 22 Dec 2022 22:21:32 -0300
Subject: [PATCH] Optimize Head Rotation


diff --git a/src/main/java/net/minecraft/server/EntityTrackerEntry.java b/src/main/java/net/minecraft/server/EntityTrackerEntry.java
index 12112ce7..5cefcd81 100644
--- a/src/main/java/net/minecraft/server/EntityTrackerEntry.java
+++ b/src/main/java/net/minecraft/server/EntityTrackerEntry.java
@@ -413,7 +413,8 @@ public class EntityTrackerEntry {
                     // CraftBukkit start - Fix for nonsensical head yaw
                     if(this.tracker instanceof EntityLiving) {
                         this.i = MathHelper.d(this.tracker.getHeadRotation() * 256.0F / 360.0F);
-                        this.broadcast(new PacketPlayOutEntityHeadRotation(this.tracker, (byte) i));
+                        // FlamePaper - Optimize Head Rotation
+                        entityplayer.playerConnection.sendPacket(new PacketPlayOutEntityHeadRotation(this.tracker, (byte) i));
                     }
                     // CraftBukkit end
 
-- 
2.35.3.windows.1

