From b6fcdca03e296ce31fe68d584626a0866f6665c0 Mon Sep 17 00:00:00 2001
From: RoccoDev <roccodev.business@gmail.com>
Date: Mon, 10 Aug 2020 11:43:26 +0200
Subject: [PATCH] Disable world saving on shutdown if auto save is disabled


diff --git a/src/main/java/net/minecraft/server/MinecraftServer.java b/src/main/java/net/minecraft/server/MinecraftServer.java
index e0742c04..8d3da09f 100644
--- a/src/main/java/net/minecraft/server/MinecraftServer.java
+++ b/src/main/java/net/minecraft/server/MinecraftServer.java
@@ -384,6 +384,9 @@ public abstract class MinecraftServer implements Runnable, ICommandListener, IAs
                 // CraftBukkit end
 
                 if (worldserver != null) {
+                    // KigPaper start - disable saving if !autoSave
+                    if(worldserver.savingDisabled) continue;
+                    // KigPaper end
                     if (!flag) {
                         MinecraftServer.LOGGER.info("Saving chunks for level \'" + worldserver.getWorldData().getName() + "\'/" + worldserver.worldProvider.getName());
                     }
-- 
2.35.1.windows.2

