From daaf8a3f34d31edc929ff23332cfcca88f241781 Mon Sep 17 00:00:00 2001
From: RoccoDev <roccodev.business@gmail.com>
Date: Mon, 10 Aug 2020 11:43:26 +0200
Subject: [PATCH] Disable world saving on shutdown if auto save is disabled


diff --git a/src/main/java/net/minecraft/server/MinecraftServer.java b/src/main/java/net/minecraft/server/MinecraftServer.java
index 9b90e325..ea38ae94 100644
--- a/src/main/java/net/minecraft/server/MinecraftServer.java
+++ b/src/main/java/net/minecraft/server/MinecraftServer.java
@@ -413,6 +413,9 @@ public abstract class MinecraftServer extends dev._2lstudios.flamepaper.tickloop
                 // CraftBukkit end
 
                 if (worldserver != null) {
+                    // KigPaper start - disable saving if !autoSave
+                    if(worldserver.savingDisabled) continue;
+                    // KigPaper end
                     if (!flag) {
                         MinecraftServer.LOGGER.info("Saving chunks for level \'" + worldserver.getWorldData().getName() + "\'/" + worldserver.worldProvider.getName());
                     }
-- 
2.35.3.windows.1

