From 876da230769fc3a35d08bea21f0ed644a80836e4 Mon Sep 17 00:00:00 2001
From: LinsaFTW <25271111+linsaftw@users.noreply.github.com>
Date: Sat, 14 Jan 2023 00:42:40 -0300
Subject: [PATCH] Fix Activation Range Double Chunk Lookup


diff --git a/src/main/java/org/spigotmc/ActivationRange.java b/src/main/java/org/spigotmc/ActivationRange.java
index c3b16c94..82588a9d 100644
--- a/src/main/java/org/spigotmc/ActivationRange.java
+++ b/src/main/java/org/spigotmc/ActivationRange.java
@@ -128,10 +128,14 @@ public class ActivationRange
             {
                 for ( int j1 = k; j1 <= l; ++j1 )
                 {
-                    if ( world.getWorld().isChunkLoaded( i1, j1 ) )
+                    // FlamePaper start - Fix Activation Range Double Chunk Lookup
+                    Chunk chunk = world.getChunkIfLoaded( i1, j1 );
+
+                    if ( chunk != null )
                     {
-                        activateChunkEntities( world.getChunkAt( i1, j1 ) );
+                        activateChunkEntities( chunk );
                     }
+                    // FlamePaper end - Fix Activation Range Double Chunk Lookup
                 }
             }
         }
-- 
2.35.3.windows.1

