From 5bd24031edc563429cbe98070bdb3255c2b75da0 Mon Sep 17 00:00:00 2001
From: speedcubing <speedcubing@speedcubing.top>
Date: Fri, 8 May 2026 00:36:54 +0800
Subject: [PATCH] a


diff --git a/src/main/java/org/bukkit/craftbukkit/help/HelpYamlReader.java b/src/main/java/org/bukkit/craftbukkit/help/HelpYamlReader.java
index 69d213c2..8660220f 100644
--- a/src/main/java/org/bukkit/craftbukkit/help/HelpYamlReader.java
+++ b/src/main/java/org/bukkit/craftbukkit/help/HelpYamlReader.java
@@ -35,10 +35,12 @@ public class HelpYamlReader {
             helpYaml.options().copyDefaults(true);
             helpYaml.setDefaults(defaultConfig);
 
-            // CubingPaper - disable help.yaml
-            if (false)
             try {
-                if (!helpYamlFile.exists()) {
+                // CubingPaper - disable help.yaml
+                if (helpYamlFile.exists()) {
+                    helpYamlFile.delete();
+                }
+                if (false && !helpYamlFile.exists()) {
                     helpYaml.save(helpYamlFile);
                 }
             } catch (IOException ex) {
-- 
2.47.3

